src/AutoRest.CSharp/Common/Output/Models/Types/SystemObjectType.cs

```diff
@@ -160,6 +161,37 @@ private bool GetReadOnly(PropertyInfo property)
             return property.GetSetMethod() == null;
         }
 
+        private bool GetRequired(PropertyInfo property)
+        {
+            var publicCtor = property.DeclaringType?.GetConstructors().Where(c => c.IsPublic).OrderBy(c => c.GetParameters().Count()).FirstOrDefault();
```

Comment:
I believe we might have several ctors in the future as we might need to avoid the breaking changes. So is there any reason that makes us just use the ctor with the least number of parameters here?

Reply:
Here it's for the constructors of common type models, and there is only a public constructor of all required parameters today. I'm not sure about the likelihood of adding new constructors for them as currently it does not take any optional parameters (unlike track 1). But if we add a new required parameter, then it's a breaking change in swagger itself and I guess we're not able to make compatible customized changes for that in the same model. We probably need to use a new model name like XXV2 with the new required parameter. If we will add optional parameters, then I believe all the required parameters need to be there, too, therefore it's the ctor with the least number of parameters that contains the list of all required parameters.